hs_core/search_indexes.py

```diff
@@ -51,49 +51,71 @@ class BaseResourceIndex(indexes.SearchIndex, indexes.Indexable):
     editors_names = indexes.MultiValueField(faceted=True)
     editors_count = indexes.IntegerField(faceted=True)
 
-
     def get_model(self):
         return BaseResource
 
     def index_queryset(self, using=None):
-        return self.get_model().objects.filter(Q(raccess__discoverable=True) | Q(raccess__public=True))
+        return self.get_model().objects.filter(Q(raccess__discoverable=True) |
+                                               Q(raccess__public=True))
 
     def prepare_title(self, obj):
-        if hasattr(obj, 'metadata'):
+        if hasattr(obj, 'metadata') and obj.metadata.title.value is not None:
             return obj.metadata.title.value
         else:
             return 'none'
 
+    def prepare_abstract(self, obj):
+        if hasattr(obj, 'metadata') and obj.metadata.description.abstract is not None:
+            return obj.metadata.description.abstract
+        else:
+            return 'none'
+
+    def prepare_author(self, obj):
+        if hasattr(obj, 'metadata'):
+            first_creator = obj.metadata.creators.filter(order=1).first()
+            if first_creator.name is not None: 
+                return first_creator.name
+            else: 
+                return 'none' 
+        else:
+            return 'none'
+
     def prepare_creators(self, obj):
-        if hasattr(obj, 'metadata'): 
-            return [creator.name for creator in obj.metadata.creators.all()]
+        if hasattr(obj, 'metadata'):
+            return [creator.name for creator in obj.metadata.creators.all()
+                    .exclude(name__isnull=True)]
         else:
             return []
 
     def prepare_contributors(self, obj):
-        if hasattr(obj, 'metadata'): 
-            return [contributor.name for contributor in obj.metadata.contributors.all()]
+        if hasattr(obj, 'metadata'):
```

Comment:
The fact that I list all organizations and all names that are not None handles this case. 

I think the only thing I probably should do is to delete the "first creator" code, because it does nothing at all. 

This document is only used for indexing, and the fact that a term appears once is enough.